File: app/src/main/java/lib/phonograph/view/SecondaryTextViewX.java
/*
 * Copyright (c) 2022 chr_56 & Abou Zeid (kabouzeid) (original author)
 */

package lib.phonograph.view;

import android.content.Context;
import android.util.AttributeSet;

import androidx.appcompat.widget.AppCompatTextView;

import util.mdcolor.pref.ThemeColor;


/**
 * @author Aidan Follestad (afollestad)
 */
public class SecondaryTextViewX extends AppCompatTextView {

    public SecondaryTextViewX(Context context) {
        super(context);
        init(context, null);
    }

    public SecondaryTextViewX(Context context, AttributeSet attrs) {
        super(context, attrs);
        init(context, attrs);
    }

    public SecondaryTextViewX(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(context, attrs);
    }

    private void init(Context context, AttributeSet attrs) {
        setTextColor(ThemeColor.textColorSecondary(context));
    }
}

File: app/src/main/java/lib/phonograph/view/SecondaryTextViewX.kt
/*
 * Copyright (c) 2022 Abou Zeid (kabouzeid) (original author)
 */
package lib.phonograph.view

import mt.util.color.secondaryTextColor
import player.phonograph.util.theme.nightMode
import androidx.appcompat.widget.AppCompatTextView
import android.content.Context
import android.util.AttributeSet

/**
 * @author Aidan Follestad (afollestad)
 */
class SecondaryTextViewX : AppCompatTextView {
    constructor(context: Context) : super(context) {
        init(context, null)
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        init(context, attrs)
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        init(context, attrs)
    }

    private fun init(context: Context, attrs: AttributeSet?) {
        setTextColor(context.secondaryTextColor(context.nightMode))
    }
}